e2e/tests/proxy_test.go

```diff
@@ -235,9 +235,26 @@ func TestProxyBadGateway(t *testing.T) {
 		"httpbin:1",
 	}
 
+	var expectedErrorMessage string
+	if os.Getenv("FORWARDER_PROXY") == "" {
+		expectedErrorMessage = "forwarder failed to connect to remote host"
+	} else {
+		expectedErrorMessage = forwarder.UpstreamProxyServiceName + " failed to connect to remote host"
+	}
+
 	for _, scheme := range []string{"http", "https"} {
 		for _, h := range hosts {
-			newClient(t, scheme+"://"+h).GET("/status/200").ExpectStatus(http.StatusBadGateway)
+			t.Run(scheme+"_"+h, func(t *testing.T) {
+				res := newClient(t, scheme+"://"+h).GET("/status/200")
+				res.ExpectStatus(http.StatusBadGateway)
+
+				// Check if the error message is correctly propagated to the client.
+				// Especially when several proxies are chained.
+				if !strings.Contains(string(res.Body), expectedErrorMessage) {
```

Comment:
Nah, just skip it if MITM is enabled.

Reply:
The issue is that our current httpexpect implementation also does the same trick, so all https requests made by the client do not return an original response in case of a connect error. 
```go
func (c *Client) do(req *http.Request) (*http.Response, error) {
	if c.trace != nil {
		req = req.WithContext(httptrace.WithClientTrace(req.Context(), c.trace))
	}
	resp, err := c.rt.RoundTrip(req)

	// There is a difference between sending HTTP and HTTPS requests.
	// For HTTPS client issues a CONNECT request to the proxy and then sends the original request.
	// In case the proxy responds with status code 4XX or 5XX to the CONNECT request, the client interprets it as URL error.
	//
	// This is to cover this case.
	if req.URL.Scheme == "https" && err != nil {
		for i := 400; i < 600; i++ {
			if err.Error() == http.StatusText(i) {
				return &http.Response{
					StatusCode: i,
					Status:     http.StatusText(i),
					ProtoMajor: 1,
					ProtoMinor: 1,
					Header:     http.Header{},
					Body:       http.NoBody,
					Request:    req,
				}, nil
			}
		}
	}

	return resp, err
}
```

Solution 1. is to keep this test and later address it in both proxy http client and our http expect client.

Solution 2. is to limit the scope and make it a dedicated MITM flag test.

Reply:
I actually wanted to print a warning there, but I'm afraid, it won't be visible in a non-debug mode. So there we have this "half skip" - it always checks the status code.

Reply:
Add text checking only for http requests.

Reply:
ok